proc-macros/tests/ui/correct/basic.rs

```diff
@@ -24,6 +24,13 @@ pub trait Rpc {
 	#[method(name = "array_params")]
 	async fn array_params(&self, items: Vec<u64>) -> RpcResult<u64>;
 
+	#[method(name = "rename_params")]
+	async fn rename_params(
+		&self,
+		#[argument(rename = "type")] r#type: u16,
```

Comment:
DQ: without this rename, would it not still be valid to pass a param with name "type"?

Reply:
I am unsure what you are asking.
I think that it would not be valid because the server would expect params with name "r#type" instead of "type".

Actually, the original usecase that got me to writing this PR was that I had a non-jsonrpsee server accepting a param "in" which could not be done through the proc-macros API because "in" is a reserved keyword (as "type" is, too) and thus all params would be serialized to "r#in", preventing the server from recognizing the param.

Reply:
So the rpc definition looks like this:

```rust
    #[method(name = "foo")]
    fn bar(f: usize, r#type: usize) -> usize {
        f + r#type
    }
```

Then in codegen we generated a struct as JSON Object with param names as the idents from the params binding such that the following would be generated:

```rust
    #[derive(serde::Serialize, serde::Deserialize)
    struct ParamsObject<G0, G1> {
        #[serde(alias = "storage_key", alias = "storageKey")]
        f: G0,
        #[serde(alias = "r_type", alias = "rType")]
        r#type: G1,
    }
```

Thus, it was not possible to make call to the server with  `type` as parameter name because it's keyword in rust prior to this PR. All good

Reply:
Aah thanks for the explanations! I thought that `r#type` might just resolve to the ident `type` anyway (ignoring the raw prefix bit that allows you to actually type it at all), but I was wrong actually about that :)